### DISCUSSION TITLE
Deduplication of files with different paths/filenames/keys

### DISCUSSION THREAD

**Discussion by shleeable** [seaweedfs/seaweedfs#5381] [Q&A]
Hello,
I'm looking to start hosting a fileserver for fediverse mastodon instances. The design would need to work as such

Every mastodon instance has it's own bucket, with it's own URL and path to serve files.
Instance 1-100 will receive a copy of a piece of media (image/video/audio) and upload it separately to their dedicated isolated bucket in a different path with a different key.

I'd like to only store 1 copy of the file and serve it to the 100 instances.

**Comment by chrislusf** [comment]
there are no dedup logic in SeaweedFS so far.

**Reply by wokalski** [reply]
Following up on this, can i manually dedup by modifying the filer store? If I have an identical file, instead of inserting it via the API, could I just create another row in postgres (my filer store) and use the same meta as the existing file? Everything will keep working, i presume?
